sqs/src/main/scala/akka/stream/alpakka/sqs/SqsModel.scala

```diff
@@ -4,11 +4,61 @@
 
 package akka.stream.alpakka.sqs
 
+import com.amazonaws.{AmazonWebServiceResult, ResponseMetadata}
 import com.amazonaws.services.sqs.model.Message
 
 import scala.compat.java8.OptionConverters._
 import scala.concurrent.duration.Duration
 
+/**
+ * Additional identifiers for FIFO messsages
+ * @see [https://docs.aws.amazon.com/AWSSimpleQueueService/latest/SQSDeveloperGuide/sqs-additional-fifo-queue-identifiers.html]
+ */
+final class FifoMessageIdentifiers(
+    val sequenceNumber: String,
+    val messageGroupId: String,
+    val messageDeduplicationId: Option[String]
+) {
+
+  override def toString: String =
+    s"FifoMessageIdentifiers(sequenceNumber=$sequenceNumber, messageGroupId=$messageGroupId, messageDeduplicationId=$messageDeduplicationId)"
+
+  override def equals(other: Any): Boolean = other match {
+    case that: FifoMessageIdentifiers =>
+      java.util.Objects.equals(this.sequenceNumber, that.sequenceNumber) &&
+      java.util.Objects.equals(this.messageGroupId, that.messageGroupId) &&
+      java.util.Objects.equals(this.messageDeduplicationId, that.messageDeduplicationId)
+    case _ => false
+  }
+
+  override def hashCode(): Int = java.util.Objects.hash(sequenceNumber, messageGroupId, messageDeduplicationId)
+}
+
+object FifoMessageIdentifiers {
+
+  /** Scala API */
+  def apply(
```

Comment:
These are not strictly required as the instances should be created within the stage, only. To allow creation for testing, we've started to add a [message factory in a `testkit` package](https://github.com/akka/alpakka/tree/master/elasticsearch/src/main/scala/akka/stream/alpakka/elasticsearch/testkit).